<a href="https://colab.research.google.com/github/pingping-0606/Python_dataviz2024/blob/main/Exam_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Exam_2

ให้สืบหาฆาตรกร ในคดี ฆาตรกรรม (murder) ที่เกิดขึ้นในวันที่ 15 มกราคม 2018 ที่เมือง SQL City โดยเริ่มจากการตรวจสอบ ข้อมูลคดี (crime_scene_report.csv)

หมายเหตุ
person.csv ข้อมูลประชากร
interview.csv ข้อมูลการสอบสวน
income.csv ข้อมูลรายได้
get_fit_now_member.csv ข้อมูลสมาชิก ฟิตเนส Get Fit Now
get_fit_now_check_in.csv ข้อมูลการเข้าใช้ ฟิตเนส Get Fit Now
facebook_event_checkin.csv ข้อมูลการเช็คอิน Facebook
driver_license.csv ข้อมูลใบขับขี่
crime_scene_report.csv ข้อมูลคดี


In [56]:
import numpy as np
import pandas as pd

In [57]:
crime_scene_report = pd.read_csv('/content/crime_scene_report.csv')

In [58]:
person = pd.read_csv('/content/person.csv')

In [59]:
interview = pd.read_csv('/content/interview.csv')

In [60]:
income = pd.read_csv('/content/income.csv')

In [61]:
get_fit_now_check_in = pd.read_csv('/content/get_fit_now_check_in.csv')

In [62]:
get_fit_now_member = pd.read_csv('/content/get_fit_now_member.csv')

In [63]:
facebook_event_checkin = pd.read_csv('/content/facebook_event_checkin.csv')

In [64]:
driver_license = pd.read_csv('/content/drivers_license.csv')

In [65]:
crime_scene_report

date     type                                        description  \
0     20180115  robbery  A Man Dressed as Spider-Man Is on a Robbery Spree   
1     20180115   murder                  Life? Dont talk to me about life.   
2     20180115   murder  Mama, I killed a man, put a gun against his he...   
3     20180215   murder                         REDACTED REDACTED REDACTED   
4     20180215   murder  Someone killed the guard! He took an arrow to ...   
...        ...      ...                                                ...   
1223  20180430  bribery                                                 \n   
1224  20180430    fraud                  ‘Why not?’ said the March Hare.\n   
1225  20180430  assault                                                 \n   
1226  20180501  assault  be NO mistake about it: it was neither more no...   
1227  20180115   murder  Security footage shows that there were 2 witne...   

              city  
0              NYC  
1           Albany  
2             Reno  
3         SQL City  
4         SQL City  
...            ...  
1223  Garden Grove  
1224         Houma  
1225       Fontana  
1226       Trenton  
1227      SQL City  

[1228 rows x 4 columns]

กรองข้อมูลที่เป็นคดีฆาตกรรม ในวันที่ 15 มกราคม 2018 ที่เมือง SQL city

In [67]:
crime_scene_report[crime_scene_report.type == 'murder']

date    type                                        description  \
1     20180115  murder                  Life? Dont talk to me about life.   
2     20180115  murder  Mama, I killed a man, put a gun against his he...   
3     20180215  murder                         REDACTED REDACTED REDACTED   
4     20180215  murder  Someone killed the guard! He took an arrow to ...   
33    20170107  murder  ‘It proves nothing of the sort!’ said Alice. ‘...   
...        ...     ...                                                ...   
1214  20180427  murder              ‘Only a thimble,’ said Alice sadly.\n   
1215  20180428  murder  wag my tail when I’m angry. Therefore I’m mad.’\n   
1220  20180429  murder  (she was obliged to say ‘creatures,’ you see, ...   
1221  20180430  murder               ‘But what am I to do?’ said Alice.\n   
1227  20180115  murder  Security footage shows that there were 2 witne...   

                city  
1             Albany  
2               Reno  
3           SQL City  
4           SQL City  
33           Memphis  
...              ...  
1214  St. Petersburg  
1215        Appleton  
1220          Toledo  
1221         Spokane  
1227        SQL City  

[148 rows x 4 columns]

In [80]:
crime_scene_report[crime_scene_report.type == 'murder'].shape[0]

148

In [78]:
crime_2018_01_15 = crime_scene_report[crime_scene_report['date'] == 20180115]
crime_2018_01_15

date     type                                        description  \
0     20180115  robbery  A Man Dressed as Spider-Man Is on a Robbery Spree   
1     20180115   murder                  Life? Dont talk to me about life.   
2     20180115   murder  Mama, I killed a man, put a gun against his he...   
5     20180115    theft                    Big Bully stole my lunch money!   
6     20180115    fraud  Lorem ipsum dolor sit amet, consectetur adipis...   
11    20180115  assault  Hamilton: Lee, do you yield? Burr: You shot hi...   
12    20180115  assault                                   Report Not Found   
972   20180115    arson  like a mouse, you know. But do cats eat bats, ...   
973   20180115  bribery  Here the other guinea-pig cheered, and was sup...   
974   20180115    fraud  The Dormouse slowly opened his eyes. ‘I wasn’t...   
975   20180115    fraud  King put on his spectacles and looked anxiousl...   
1227  20180115   murder  Security footage shows that there were 2 witne...   

                  city  
0                  NYC  
1               Albany  
2                 Reno  
5              Chicago  
6              Seattle  
11            SQL City  
12            SQL City  
972   San Buenaventura  
973         Marysville  
974             Yakima  
975              Bryan  
1227          SQL City

In [82]:
crime_2018_01_15 = crime_scene_report[crime_scene_report['date'] == 20180115].shape[0]
crime_2018_01_15

12

In [69]:
crime_scene_report[crime_scene_report.city == 'SQL City']

date     type                                        description  \
3     20180215   murder                         REDACTED REDACTED REDACTED   
4     20180215   murder  Someone killed the guard! He took an arrow to ...   
7     20170712    theft  A lone hunter stalks the night, firing arrows ...   
8     20170820    arson  Wield the Hammer of Sol with honor, Titan, it ...   
9     20171110  robbery  The Gjallarhorn shoulder-mounted rocket system...   
10    20180103  bribery  Apparently, Cayde thought it necessary to expo...   
11    20180115  assault  Hamilton: Lee, do you yield? Burr: You shot hi...   
12    20180115  assault                                   Report Not Found   
1227  20180115   murder  Security footage shows that there were 2 witne...   

          city  
3     SQL City  
4     SQL City  
7     SQL City  
8     SQL City  
9     SQL City  
10    SQL City  
11    SQL City  
12    SQL City  
1227  SQL City

In [83]:
crime_scene_report[crime_scene_report.city == 'SQL City'].shape[0]

9

เหตูการณ์ที่เกิดขึ้นในวันที่ 15 มกราคม 2018

In [84]:
crime_scene_report[(crime_scene_report['date']==20180115) & (crime_scene_report['type']=='murder') & (crime_scene_report['city']=='SQL City')]


date    type                                        description  \
1227  20180115  murder  Security footage shows that there were 2 witne...   

          city  
1227  SQL City

คำอธิบายเหตุฆาตกรรมที่เกิดขึ้นในวันที่ 15 มกราคม 2018 ที่เมือง SQL คือ ภาพจากกล้องวงจรปิดพบว่ามีพยาน 2 คน พยานคนแรกอาศัยอยู่ที่บ้านหลังสุดท้ายบน Northwestern Dr พยานคนที่สองชื่อแอนนาเบล อาศัยอยู่ที่ไหนสักแห่งบน Franklin Ave

ต่อมาจะทำการหาข้อมูลอื่นๆของ พยานคนแรกและคนที่สอง

In [87]:
person

id                 name  license_id  address_number  \
0      10000   Christoper Peteuil      993845             624   
1      10007  Kourtney Calderwood      861794            2791   
2      10010            Muoi Cary      385336             741   
3      10016          Era Moselle      431897            1987   
4      10025         Trena Hornby      550890             276   
...      ...                  ...         ...             ...   
10006  99936          Luba Benser      274427             680   
10007  99941      Roxana Mckimley      975942            1613   
10008  99965      Cherie Zeimantz      287627            3661   
10009  99982          Allen Cruse      251350            3126   
10010  99990         Vance Hunten      830407            3056   

      address_street_name        ssn  
0            Bankhall Ave  747714076  
1           Gustavus Blvd  477972044  
2         Northwestern Dr  828638512  
3           Wood Glade St  614621061  
4           Daws Hill Way  223877684  
...                   ...        ...  
10006        Carnage Blvd  685095054  
10007             Gate St  512136801  
10008       The Water Ave  362877324  
10009           N Jean Dr  348734531  
10010       Lancefield St  896677562  

[10011 rows x 6 columns]

หาว่ามีใครบ้างที่อาศัยอยู่บน Northwestern Dr

In [88]:
person[person.address_street_name == 'Northwestern Dr']

id                  name  license_id  address_number  \
2     10010             Muoi Cary      385336             741   
282   12711        Norman Apolito      667757             599   
499   14887        Morty Schapiro      118009            4919   
528   15171          Weldon Penso      336999             311   
811   17729        Lasonya Wildey      439686            3824   
904   18376              Josh Shi      653712            1091   
1021  19420           Cody Schiel      890431            3524   
1338  22239         Dusty Sigafus      710517            1125   
1426  23044          Val Portlock      924989            3143   
1522  23960  Kristopher Lagerberg      658777            1392   
1686  25615         Olevia Morena      388477             812   
1812  26758      Angelena Billman      942074             629   
1981  28360        Rashad Cascone      924517            3212   
2131  29608       Willetta Lurvey      800568            1028   
2535  33415         Homer Simpson      108392             981   
2620  34155      Allyson Lazenson      814209            2840   
2635  34352    Vincenza Burkhardt      436470            2512   
2871  36378        Freddie Ellzey      267882            3449   
3237  39688            Abe Roeker      537445             689   
3316  40336       Fredia Billeter      251082             404   
3368  40760          Renata Faver      976650             745   
3743  44004           Alison Eska      327797            2951   
3795  44542        Ivory Mcgrevey      873842             897   
4595  51114     Christena Saffell      814284            3055   
4643  51552           Bonita Knie      764124            1726   
4806  53076           Boris Bijou      664914            3327   
4871  53699      Christian Pagano      628433            2585   
4886  53890        Sophie Tiberio      957671            3755   
4956  54564         Sammie Sengun      217286            2590   
5024  55130        Shiloh Huntoon      472118             951   
5450  59307          Cyril Yongue      753524            2903   
6476  68690            Yer Modest      852404            3046   
6698  70485     Andera Kostelecky      354822            1252   
6865  71924        Laurie Scronce      774116            1418   
6877  72076     Courtney Bordeaux      324614            1707   
7040  73368        Torie Thalmann      773862            3697   
7296  75484          Del Tacderen      447887             607   
7888  80921       Donovan Yackley      222395            2386   
7902  81052     Calvin Czelusniak      471483            1516   
7993  81885         Kurtis Masuda      934201            2108   
8369  85280             Donn Duin      569122             540   
8608  87456   Leonora Wolfsberger      215868            3483   
8720  88423           Teri Ehrich      510876            1879   
8888  89906       Kinsey Erickson      510019             309   
8939  90397     Dionna Kranwinkle      797260            2678   
9272  93509        Emmitt Aceuedo      916706            3491   
9569  96178           Galina Khuu      450248            2183   
9609  96595         Coretta Cubie      303645            3631   
9841  98593           Jonah Toner      578982            2947   
9991  99826          Ivy Kazarian      994470             892   

     address_street_name        ssn  
2        Northwestern Dr  828638512  
282      Northwestern Dr  778264744  
499      Northwestern Dr  111564949  
528      Northwestern Dr  131379495  
811      Northwestern Dr  917817122  
904      Northwestern Dr  193899001  
1021     Northwestern Dr  947110049  
1338     Northwestern Dr  724386723  
1426     Northwestern Dr  100593316  
1522     Northwestern Dr  492912529  
1686     Northwestern Dr  962831331  
1812     Northwestern Dr  341854756  
1981     Northwestern Dr  838907424  
2131     Northwestern Dr  843751378  
2535     Northwestern Dr  871392019  
2620     Northwestern Dr  991150469  
2635     Northwestern Dr  197913046  
2871     

เรียง address_number จากน้อยไปมาก เพื่อหาว่าบ้านหลังไหนคือหลังสุดท้าย

In [94]:
person[person.address_street_name == 'Northwestern Dr'].sort_values('address_number')


id                  name  license_id  address_number  \
8888  89906       Kinsey Erickson      510019             309   
528   15171          Weldon Penso      336999             311   
3316  40336       Fredia Billeter      251082             404   
8369  85280             Donn Duin      569122             540   
282   12711        Norman Apolito      667757             599   
7296  75484          Del Tacderen      447887             607   
1812  26758      Angelena Billman      942074             629   
3237  39688            Abe Roeker      537445             689   
2     10010             Muoi Cary      385336             741   
3368  40760          Renata Faver      976650             745   
1686  25615         Olevia Morena      388477             812   
9991  99826          Ivy Kazarian      994470             892   
3795  44542        Ivory Mcgrevey      873842             897   
5024  55130        Shiloh Huntoon      472118             951   
2535  33415         Homer Simpson      108392             981   
2131  29608       Willetta Lurvey      800568            1028   
904   18376              Josh Shi      653712            1091   
1338  22239         Dusty Sigafus      710517            1125   
6698  70485     Andera Kostelecky      354822            1252   
1522  23960  Kristopher Lagerberg      658777            1392   
6865  71924        Laurie Scronce      774116            1418   
7902  81052     Calvin Czelusniak      471483            1516   
6877  72076     Courtney Bordeaux      324614            1707   
4643  51552           Bonita Knie      764124            1726   
8720  88423           Teri Ehrich      510876            1879   
7993  81885         Kurtis Masuda      934201            2108   
9569  96178           Galina Khuu      450248            2183   
7888  80921       Donovan Yackley      222395            2386   
2635  34352    Vincenza Burkhardt      436470            2512   
4871  53699      Christian Pagano      628433            2585   
4956  54564         Sammie Sengun      217286            2590   
8939  90397     Dionna Kranwinkle      797260            2678   
2620  34155      Allyson Lazenson      814209            2840   
5450  59307          Cyril Yongue      753524            2903   
9841  98593           Jonah Toner      578982            2947   
3743  44004           Alison Eska      327797            2951   
6476  68690            Yer Modest      852404            3046   
4595  51114     Christena Saffell      814284            3055   
1426  23044          Val Portlock      924989            3143   
1981  28360        Rashad Cascone      924517            3212   
4806  53076           Boris Bijou      664914            3327   
2871  36378        Freddie Ellzey      267882            3449   
8608  87456   Leonora Wolfsberger      215868            3483   
9272  93509        Emmitt Aceuedo      916706            3491   
1021  19420           Cody Schiel      890431            3524   
9609  96595         Coretta Cubie      303645            3631   
7040  73368        Torie Thalmann      773862            3697   
4886  53890        Sophie Tiberio      957671            3755   
811   17729        Lasonya Wildey      439686            3824   
499   14887        Morty Schapiro      118009            4919   

     address_street_name        ssn  
8888     Northwestern Dr  635287661  
528      Northwestern Dr  131379495  
3316     Northwestern Dr  108056488  
8369     Northwestern Dr  613873810  
282      Northwestern Dr  778264744  
7296     Northwestern Dr  243821337  
1812     Northwestern Dr  341854756  
3237     Northwestern Dr  438969447  
2        Northwestern Dr  828638512  
3368     Northwestern Dr  547572040  
1686     Northwestern Dr  962831331  
9991     Northwestern Dr  208359813  
3795     Northwestern Dr  988557407  
5024     Northwestern Dr  826647644  
2535     Northwestern Dr  871392019  
2131     Northwestern Dr  843751378  
904      Northwestern Dr  193899001  
1338     

พยานคนแรกคือ Morty Schapiro เลขประจำตัวคือ 14887 มีเลขที่ใบขับขี่คือ 118009 อยู่บ้านเลขที่ 4919  ssn คือ 111564949

In [97]:
person[person.address_street_name == 'Franklin Ave']

id                 name  license_id  address_number  \
235   12207       Wilmer Wolever      509484             139   
665   16371       Annabel Miller      490173             103   
807   17683        Johnnie Schee      968887            1277   
936   18651        Carleen Etoll      356746              22   
1377  22636       Zachary Ybarbo      768359             785   
1608  24737           Gema Nantz      273410            3968   
2236  30654      Clarita Rickels      418084            2254   
2427  32264       Shelby Dezeeuw      735415            1391   
2584  33793         Amado Mattan      161915              99   
2661  34592    Cordell Lindamood      592762            3657   
2864  36328         Domenic Laun      971080            3809   
2877  36400         Noe Pancoast      895755            3443   
3009  37616     Katelyn Schwerin      430143             332   
4086  46827         Bev Billiter      643591            2316   
4708  52059        Jordan Pelton      686854            2786   
4709  52075             Ezra Phy      928109             384   
4714  52121       Yelena Whitler      427830            3185   
4990  54817          Guy Bustard      213929             591   
4998  54896        Patty Wamsley      688488            1097   
5142  56142       William Shiver      741336            1926   
5233  57051        Lavette Koehl      137365            2284   
5497  59746        Ronny Gumbert      469722            3197   
5641  60944        Ricki Bidding      443364            2072   
5653  61001      Laurine Bousman      197150             247   
5709  61437     Colette Hollomon      661377             453   
5810  62292       Isaiah Holsten      913856             747   
5829  62520      Mariah Lepetich      450447            3086   
6326  67292       Renita Roperto      801041             625   
6597  69719        Hope Arzabala      311724             841   
6641  70050   Whitney Herkenratt      138462            3445   
7300  75510  Sebastian Ramnarase      333188            1127   
7357  76128         Ressie Razze      633497            1199   
7574  78170          Graig Summy      192325            3387   
7610  78408         Dante Eschen      218173            3447   
7637  78658        Blake Chrones      310242            2014   
7658  78830        Candice Train      814779            3255   
8120  83003       Wilmer Casella      672050            3564   
8203  83754          Maria Walsh      201334            2716   
8302  84531       Edgar Mendieta      315790            3881   
8624  87521          Moses Ikerd      638991            2283   
8783  88966            Raul Eads      570927            2066   
9322  93947        Omer Andreoni      605718              13   
9462  95119            Hong Lisa      825828             375   
9756  97913       Cameron Dilick      971988            2954   
9864  98744        Jordan Myntti      256116            3104   

     address_street_name        ssn  
235         Franklin Ave  636825374  
665         Franklin Ave  318771143  
807         Franklin Ave  815977821  
936         Franklin Ave  193369255  
1377        Franklin Ave  285346605  
1608        Franklin Ave  180545802  
2236        Franklin Ave  714941023  
2427        Franklin Ave  143197463  
2584        Franklin Ave  125205748  
2661        Franklin Ave  509890333  
2864        Franklin Ave  825329637  
2877        Franklin Ave  732212073  
3009        Franklin Ave  392365033  
4086        Franklin Ave  940192302  
4708        Franklin Ave  676224073  
4709        Franklin Ave  176404498  
4714        Franklin Ave  115106155  
4990        Franklin Ave  820850670  
4998        Franklin Ave  725272037  
5142        Franklin Ave  219913461  
5233        Franklin Ave  776046786  
5497        Franklin Ave  561067477  
5641        Franklin Ave  614976330  
5653        Franklin Ave  431360364  
5709        Franklin Ave  314956038  
5810        Franklin Ave  995368430  
5829        Franklin Ave  797745632  


In [99]:
person[(person.name == 'Annabel Miller') & (person.address_street_name == 'Franklin Ave')]

id            name  license_id  address_number address_street_name  \
665  16371  Annabel Miller      490173             103        Franklin Ave   

           ssn  
665  318771143

พยานคนที่สองคือ Annabel Miller เลขประจำตัวคือ 16371 มีเลขที่ใบขับขี่คือ 490173 อยู่บ้านเลขที่ 103  ssn คือ 318771143

In [100]:
interview

person_id                                         transcript
0         28508                ‘I deny it!’ said the March Hare.\n
1         63713                                                 \n
2         86208      way, and the whole party swam to the shore.\n
3         35267  lessons in here? Why, there’s hardly room for ...
4         33856                                                 \n
...         ...                                                ...
4986      37357  Alice did not wish to offend the Dormouse agai...
4987      10206  time,’ she said, ‘than waste it in asking ridd...
4988      14887  I heard a gunshot and then saw a man run out. ...
4989      16371  I saw the murder happen, and I recognized the ...
4990      67318  I was hired by a woman with a lot of money. I ...

[4991 rows x 2 columns]

In [102]:
interview[(interview.person_id == 14887) | (interview.person_id == 16371)]


person_id                                         transcript
4988      14887  I heard a gunshot and then saw a man run out. ...
4989      16371  I saw the murder happen, and I recognized the ...

ข้อมูลการสอบสวนของพยานคนแรกคือ ฉันได้ยินเสียงปืนดังขึ้นและเห็นชายคนหนึ่งวิ่งออกมา เขาถือกระเป๋า "Get Fit Now Gym" หมายเลขสมาชิกบนกระเป๋าเริ่มต้นด้วย "48Z" เฉพาะสมาชิกโกลด์เท่านั้นที่มีกระเป๋าแบบนั้น ชายคนนั้นขึ้นรถที่มีป้ายทะเบียน "H42W"

ข้อมูลการสอบสวนของพยานคนที่สองคือ ฉันได้เห็นการฆาตกรรมเกิดขึ้น และฉันจำฆาตกรได้จากโรงยิมของฉัน ขณะที่ฉันกำลังออกกำลังกายเมื่อสัปดาห์ที่แล้วในวันที่ 9 มกราคม

ต่อมาจะทำการหาว่าใครที่เป็นสมาชิกโกลด์ของยิมนี้ ที่มีหมายเลขสมาชิกบนกระเป๋าเริ่มต้นด้วย "48Z" และป้ายทะเบียนรถเป็น "H42W"

In [103]:
get_fit_now_member

id  person_id                name  membership_start_date  \
0    NL318      65076     Everette Koepke               20170926   
1    AOE21      39426        Noe Locascio               20171005   
2    2PN28      63823  Jeromy Heitschmidt               20180215   
3    0YJ24      80651      Waneta Wellard               20171206   
4    3A08L      32858        Mei Bianchin               20170401   
..     ...        ...                 ...                    ...   
179  2V137      41693      Wendell Dulany               20171219   
180  4KB72      79110          Emile Hege               20170522   
181  48Z7A      28819        Joe Germuska               20160305   
182  48Z55      67318       Jeremy Bowers               20160101   
183  90081      16371      Annabel Miller               20160208   

    membership_status  
0                gold  
1             regular  
2              silver  
3                gold  
4              silver  
..                ...  
179            silver  
180           regular  
181              gold  
182              gold  
183              gold  

[184 rows x 5 columns]

In [105]:
get_fit_now_member[(get_fit_now_member.membership_status == 'gold') & (get_fit_now_member.id.str[:3] == '48Z')]


id  person_id           name  membership_start_date membership_status
181  48Z7A      28819   Joe Germuska               20160305              gold
182  48Z55      67318  Jeremy Bowers               20160101              gold

จากคำให้การของพยานคนแรกมีผู้ต้องสงสัยสองคนคือ Joe Germuska และ Jeremy Bowers

In [108]:
driver_license

id  age  height eye_color hair_color  gender plate_number  \
0      100280   72      57     brown        red    male       P24L4U   
1      100460   63      72     brown      brown  female       XF02T6   
2      101029   62      74     green      green  female       VKY5KR   
3      101198   43      54     amber      brown  female       Y5NZ08   
4      101255   18      79      blue       grey  female       5162Z1   
...       ...  ...     ...       ...        ...     ...          ...   
10002  999923   19      77     amber      black  female       5L0ZI4   
10003  999940   71      61     green      green    male       1B8QN8   
10004  999981   67      69     brown       blue  female       1684K3   
10005  999986   49      58     green       grey    male       F8F64H   
10006  999993   18      63     black      black  female       6UZO2O   

         car_make    car_model  
0           Acura          MDX  
1        Cadillac          SRX  
2           Scion           xB  
3          Nissan        Rogue  
4           Lexus           GS  
...           ...          ...  
10002         GMC  Sierra 3500  
10003  Mitsubishi      Eclipse  
10004  Land Rover          LR2  
10005       Lexus           LS  
10006    Cadillac      DeVille  

[10007 rows x 9 columns]

In [120]:
driver_license[driver_license.plate_number.str[:4] == 'H42W']

id  age  height eye_color hair_color  gender plate_number car_make  \
915  183779   21      65      blue     blonde  female       H42W0X   Toyota   

    car_model  
915     Prius

จากคำให้การของพยานคนแรกว่าเห็นฆาตกรขึ้นรถทะเบียน H42W ซึ่งเจ้าของรถเป็นผู้หญิง อายุ 21 ปี ผมสีบลอนด์ ตาสีฟ้า

In [121]:
get_fit_now_check_in

membership_id  check_in_date  check_in_time  check_out_time
0            NL318       20180212            329             365
1            NL318       20170811            469             920
2            NL318       20180429            506             554
3            NL318       20180128            124             759
4            NL318       20171027            418            1019
...            ...            ...            ...             ...
2698         4KB72       20170422           1016            1114
2699         4KB72       20170630            408             885
2700         48Z7A       20180109           1600            1730
2701         48Z55       20180109           1530            1700
2702         90081       20180109           1600            1700

[2703 rows x 4 columns]

In [128]:
get_fit_now_check_in[(get_fit_now_check_in.check_in_date == 20180109) & (get_fit_now_check_in.membership_id.str[:3] == '48Z')]


membership_id  check_in_date  check_in_time  check_out_time
2700         48Z7A       20180109           1600            1730
2701         48Z55       20180109           1530            1700

จากคำให้การของพยานคนที่สองว่าฆาตกรเป็นผู้ที่เข้าใช้บริการยิมในวันที่ 9 มกราคม 2018 มีผู้ต้องสงสัยตรงกับพยานคนแรกคือ Joe Germuska และ Jeremy Bowers

In [131]:
person[(person.name == 'Joe Germuska') | (person.name == 'Jeremy Bowers')]

id           name  license_id  address_number    address_street_name  \
2037  28819   Joe Germuska      173289             111                Fisk Rd   
6327  67318  Jeremy Bowers      423327             530  Washington Pl, Apt 3A   

            ssn  
2037  138909730  
6327  871539279

In [129]:
facebook_event_checkin

person_id  event_id                                         event_name  \
0          28508      5880    Nudists are people who wear one-button suits.\n   
1          63713      3865  but that's because it's the best book on anyth...   
2          63713      3999           If Murphy's Law can go wrong, it will.\n   
3          63713      6436  Old programmers never die.  They just branch t...   
4          82998      4470               Help a swallow land at Capistrano.\n   
...          ...       ...                                                ...   
20006      99716      1143                               SQL Symphony Concert   
20007      99716      1143                               SQL Symphony Concert   
20008      99716      1143                               SQL Symphony Concert   
20009      67318      4719                             The Funky Grooves Tour   
20010      67318      1143                               SQL Symphony Concert   

           date  
0      20170913  
1      20171009  
2      20170502  
3      20170926  
4      20171022  
...         ...  
20006  20171206  
20007  20171212  
20008  20171229  
20009  20180115  
20010  20171206  

[20011 rows x 4 columns]

In [134]:
facebook_event_checkin[facebook_event_checkin.person_id == 28819]


Empty DataFrame
Columns: [person_id, event_id, event_name, date]
Index: []

Joe Germuska ไม่มีข้อมูล facebook_event_checkin

In [135]:
facebook_event_checkin[facebook_event_checkin.person_id == 67318]

person_id  event_id              event_name      date
20009      67318      4719  The Funky Grooves Tour  20180115
20010      67318      1143    SQL Symphony Concert  20171206

Jeremy Bowers มีข้อมูลการค้นหาคือ ทัวร์ Funky Grooves ในวันที่ 15 มกราคม 2018 และ คอนเสิร์ต SQL Symphony ในวันที่ 6 ธันวาคม 2017

In [137]:
interview[interview.person_id == 67318]

person_id                                         transcript
4990      67318  I was hired by a woman with a lot of money. I ...

คำให้การของ Jeremy Bowers คือ ฉันถูกจ้างโดยผู้หญิงคนหนึ่งที่มีเงินมากมาย ฉันไม่ทราบชื่อของเธอ แต่ฉันรู้ว่าเธอสูงประมาณ 5 ฟุต 5 นิ้ว (65 นิ้ว) หรือ 5 ฟุต 7 นิ้ว (67 นิ้ว) เธอมีผมสีแดงและขับรถ Tesla Model S ฉันรู้ว่าเธอไปชมคอนเสิร์ต SQL Symphony 3 ครั้งในเดือนธันวาคม 2017

สรุปได้ว่าฆาตกรคือ Jeremy Bowers เป็นผู้ลงมือสังหาร โดยมีผู้จ้างอีกทีคาดว่าเขาน่าจะได้เจอกับเธอตอนที่ไปชมคอนเสิร์ต SQL Symphonyในเดือนธันวาคม ปี 2017 เพราะประวัติการค้นหา facebook ของฆาตกร มีการค้นหา คอนเสิร์ต SQL Symphony ในวันที่ 6 ธันวาคม 2017 ซึ่งก็สามารถเป็นหลักฐานได้ว่าฆาตกรได้เจอกับคนจ้างวานจริงๆ